### DISCUSSION TITLE
Custom script option to import entire data source

### DISCUSSION THREAD

**Discussion by ghost** [netbox-community/netbox#14123] [Ideas]
This will likely depend on #12431 but in our use case we have a few sets of repos with a collection of scripts and supporting files we import directly to our custom Netbox container.

While trying to move towards the data source import model we can't import the whole data source directory similar to running a git clone command within the /etc/netbox/scripts directory.
Since some repos are 100+ files, continuously adding each file into scripts is not maintainable, we should have an additional option to import all files within a data source and keep them in sync instead of only a single file at a time.

![image](https://github.com/netbox-community/netbox/assets/83660592/12b9d1c2-78d5-47a0-af68-a8035341b59a)

**Comment by wifidoneright** [comment]
i am part of a very large organization and this feature would allow for much more comprehensive scripts that also share functionality. requiring everything that a script does to be within a single file does not allow for scaling the custom scripts.
